src/services/user.service.ts

```diff
@@ -21,4 +22,65 @@ export class UserService implements IUserService {
         return await UserModel.insertMany([userRegisterDTO]);
     }
 
+    public async login(userLoginDTO: UserLoginDTO): Promise<UserLoginResponseDTO> {
+        const user = await UserModel.findOne({email: userLoginDTO.email});
+
+        if (!user?._id) {
+            throw new Unauthorized('wrong email or password');
+        }
+
+        if (!await compare(userLoginDTO.password, user?.password || '')) {
+            throw new Unauthorized('wrong email or password');
+        }
+
+        const token = await generateToken(user?._id, Date.now() / 1000 + config.exp)
+        const userLoginResponseDTO: UserLoginResponseDTO =  {
+            _id: user?._id,
+            token
+        }
+
+        await UserModel.updateOne(
+            {email: user.email},
+            {token}
+        );
+
+        return userLoginResponseDTO;
+    }
+
+    public async edit(userEditDTO: UserEditDTO): Promise<void> {
+        const anyUser = await UserModel.findOne({email: userEditDTO.newEmail})
+        if (anyUser?.email) {
+            throw new ValueAlreadyInUse('new email is already taken');
+        }
+
+        const user = await UserModel.findOne({email: userEditDTO.email})
+        if (!user?.email) {
+            throw new Unauthorized('user with the email does not exist');
+        }
+
+        userEditDTO.newPassword = await hash(userEditDTO.newPassword);
+
+        await UserModel.updateOne({email: userEditDTO.email}, {
+            email: userEditDTO.newEmail,
+            password: userEditDTO.newPassword,
+        })
+    }
+
+    public async verifyAndRestartToken(id: string, token: string): Promise<void> {
```

Comment:
never create functions with and. create two functions one for verifying and the second one for the restart

Reply:
agreed, updated. however I have a general question related to it. (posted as a code comment)